src/main/java/at/ac/tuwien/kr/alpha/solver/learning/GroundConflictNoGoodLearner.java

```diff
@@ -110,174 +117,144 @@ public GroundConflictNoGoodLearner(Assignment assignment) {
-				return new ConflictAnalysisResult(null, atomAssignmentEntry.getWeakDecisionLevel(), true, resolutionAtoms);
+	private int backjumpLevelRemovingConflict(Antecedent violatedNoGood) {
+		int highestDL = 0;
+		int[] reasonLiterals = violatedNoGood.getReasonLiterals();
+		for (int literal : reasonLiterals) {
+			int literalDL = assignment.getWeakDecisionLevel(atomOf(literal)); //Math.min(assignment.getWeakDecisionLevel(atomOf(literal)), ((TrailAssignment) assignment).getOutOfOrderDecisionLevel(atomOf(literal)));
+			if (literalDL > highestDL) {
+				highestDL = literalDL;
 			}
-			// Case a) take other implying NoGood into account.
-			int resolutionAtom = atomAssignmentEntry.getAtom();
-			resolutionAtoms.add(resolutionAtom);
-			currentResolutionNoGood = NoGood.learnt(resolveNoGoods(firstUIPPriorityQueue, currentResolutionNoGood, otherContributingNoGood, resolutionAtom, atomAssignmentEntry.getWeakDecisionLevel(), atomAssignmentEntry.getPropagationLevelRespectingLowerMBT()));
-			
-			// TODO: create/edit ResolutionSequence
 		}
+		return highestDL - 1;
+	}
 
-		while (true) {
-			// Check if 1UIP was reached.
-			if (firstUIPPriorityQueue.size() == 1) {
-				// Only one remaining literals to process, we reached 1UIP.
-				int backjumpingDecisionLevel = computeBackjumpingDecisionLevel(currentResolutionNoGood);
-				if (backjumpingDecisionLevel < 0) {
-					return repeatAnalysisIfNotAssigning(currentResolutionNoGood, resolutionAtoms);
-				}
-				return new ConflictAnalysisResult(currentResolutionNoGood, backjumpingDecisionLevel, false, resolutionAtoms);
-			} else if (firstUIPPriorityQueue.size() < 1) {
-				// This can happen if some NoGood implied a literal at a higher decision level and later the implying literals become (re-)assigned at lower decision levels.
-				// Lowering the decision level may be possible but requires further analysis.
-				// For the moment, just report the learned NoGood.
-				return repeatAnalysisIfNotAssigning(currentResolutionNoGood, resolutionAtoms);
-				//return new ConflictAnalysisResult(currentResolutionNoGood, computeBackjumpingDecisionLevel(currentResolutionNoGood), false, noGoodsResponsible);
-			} else if (currentResolutionNoGood.size() > 32) {
-				// Break if resolved NoGood becomes too large.
-				// Remove all current-dl elements from the resolution NoGood and add the last choice, then backjump like usual.
-				return new ConflictAnalysisResult(null, conflictDecisionLevel, true, resolutionAtoms);	// Flag unsatisfiable abused here.
-				/*if (getAssignmentEntryRespectingLowerMBT(lastGuessedAtom).getDecisionLevel() <= conflictDecisionLevel) {
-					// If lastGuessedAtom is not unassigned after backjump, use repeatAnalysisIfNotAssigning.
-					return repeatAnalysisIfNotAssigning(replaceAllFromConflictDecisionLevelWithGuess(currentResolutionNoGood, conflictDecisionLevel, lastGuessedAtom), noGoodsResponsible, lastGuessedAtom);
+	private String printReasons(Collection<Integer> reasons) {
```

Comment:
This method doesn´t actually _print_ anything, so I propose renaming it to `reasonsToString`
```suggestion
	private String reasonsToString(Collection<Integer> reasons) {
```